src/stores/game.js

```diff
@@ -806,5 +826,19 @@ export const useGameStore = defineStore('game', {
         });
       });
     },
+    async requestRematch({ gameId, rematch = true }) {
+      return new Promise((resolve, reject) => {
+        io.socket.get('/game/rematch', { gameId, rematch }, (res, jwres) => {
+          return this.handleGameResponse(jwres, resolve, reject);
+        });
+      });
+    },
+    async requestJoinRematch({ oldGameId }) {
+      return await new Promise((resolve, reject) => {
```

Comment:
We generally don't need to await the returned promise
```suggestion
      return new Promise((resolve, reject) => {
```